typescript/sdk/src/middleware/account/InterchainAccount.ts

```diff
@@ -139,20 +147,40 @@ export class InterchainAccount extends RouterApp<InterchainAccountFactories> {
 
   // general helper for different overloaded callRemote functions
   // can override the gasLimit by StandardHookMetadata.overrideGasLimit for optional hookMetadata here
-  async callRemote(
-    chain: ChainName,
-    destination: ChainName,
-    calls: Array<CallData>,
-    config: AccountConfig,
-    hookMetadata?: string,
-  ): Promise<void> {
+  async callRemote({
+    chain,
+    destination,
+    innerCalls,
+    config,
+    hookMetadata,
+  }: GetCallRemoteSettings): Promise<void> {
     await this.multiProvider.sendTransaction(
       chain,
-      this.getCallRemote(chain, destination, calls, config, hookMetadata),
+      this.getCallRemote({
+        chain,
+        destination,
+        innerCalls,
+        config,
+        hookMetadata,
+      }),
     );
   }
 }
 
+export function getInterchainAccount(
```

Comment:
as is this suggests that we are actually computing the interchain account address from the config, not building the app
maybe `buildInterchainAccountApp`?